zamba/object_detection/yolox/megadetector_lite_yolox.py

```diff
@@ -116,19 +122,63 @@ def _preprocess(self, frame: np.ndarray) -> np.ndarray:
         """Process an image for the model, including scaling/padding the image, transposing from
         (height, width, channel) to (channel, height, width) and casting to float.
         """
-        return np.ascontiguousarray(
-            self.scale_and_pad_array(
-                frame, self.config.image_width, self.config.image_height
-            ).transpose(2, 0, 1),
+        arr = np.ascontiguousarray(
+            self.scale_and_pad_array(frame, self.config.image_width, self.config.image_height),
             dtype=np.float32,
         )
+        return np.moveaxis(arr, 2, 0)
+
+    def _preprocess_video(self, video: np.ndarray) -> np.ndarray:
+        """Process a video for the model, including resizing the frames in the video, transposing
+        from (batch, height, width, channel) to (batch, channel, height, width) and casting to float.
+        """
+        resized_frames = []
+        for frame_idx in range(video.shape[0]):
+            frame = video[frame_idx]
+            resized_frames.append(self._preprocess(frame))
+
+        return np.array(resized_frames)
```

Comment:
This will be inefficient memory-wise since we store as a Python list and then rebuild the array. Likely better and faster to pre-allocate with `np.zeros` and then fill this in with what is returned from `self._preprocess`

Reply:
ah, great flag. thanks